--- src/crc32.c.orig	2025-05-12 13:07:18
+++ src/crc32.c	2025-05-12 13:07:39
@@ -132,12 +132,12 @@
 /* =========================================================================
  * This function can be used by asm versions of crc32()
  */
-const uLongf * ZEXPORT get_crc_table()
+const z_crc_t FAR * ZEXPORT get_crc_table()
 {
 #ifdef DYNAMIC_CRC_TABLE
   if (crc_table_empty) make_crc_table();
 #endif
-  return (const uLongf *)crc_table;
+  return (const z_crc_t *)crc_table;
 }
 
 /* ========================================================================= */
